docs/website/docs/walkthroughs/deploy-a-pipeline/deploy-with-modal.md

```diff
@@ -0,0 +1,154 @@
+- Cloud Functions: Run Python code in the cloud instantly and scale horizontally.
+- GPU/CPU Scaling: Easily attach GPUs for heavy tasks like AI model training with a single line of code.
+- Web Endpoints: Expose any function as an HTTPS API endpoint quickly.
+- Scheduled Jobs: Convert Python functions into scheduled tasks effortlessly.
+
+To know more, please refer to [Modals's documentation.](https://modal.com/docs)
+
+## Building data pipelines with dlt
+
+dlt is an open-source Python library that allows you to declaratively load data sources into well-structured tables or datasets. It does this through automatic schema inference and evolution. The library simplifies building data pipelines by providing functionality to support the entire extract and load process.
+
+### How does dlt integrate with Modal for pipeline orchestration?
+
+As an example of how to set up a pipeline in Modal, we'll use the [building a cost-effective analytics stack with Modal, dlt, and dbt.](https://modal.com/blog/analytics-stack) case study.
+
+The example demonstrates automating a workflow to load data from Postgres to Snowflake using `dlt`.
+
+## How to run dlt on Modal
+
+Here’s our `dlt` setup copying data from our Postgres read replica into Snowflake:
+
+1. Run the `dlt` SQL database setup to initialize their `sql_database_pipeline.py` template:
+   ```sh
+   dlt init sql_database snowflake
+   ```
+2. Open the file and define the Modal Image you want to run `dlt` in:
+   ```py
+   import dlt
+   import pendulum
+
+   from sql_database import sql_database, ConnectionStringCredentials, sql_table
+
+   import modal
+   import os
+
+   image = (
+       modal.Image.debian_slim()
+       .apt_install(["libpq-dev"]) # system requirement for postgres driver
+       .pip_install(
+           "sqlalchemy>=1.4", # how `dlt` establishes connections
+           "dlt[snowflake]>=0.4.11",
+           "psycopg2-binary", # postgres driver
+           "dlt[parquet]",
+           "psutil==6.0.0", # for `dlt` logging
+           "connectorx", # creates arrow tables from database for fast data extraction
+       )
+   )
+
+   app = modal.App("dlt-postgres-pipeline", image=image)
+   ```
+
+3. Wrap the provided `load_table_from_database` with the Modal Function decorator, Modal Secrets containing your database credentials, and a daily cron schedule
+   ```py
+   # Function to load the table from the database, scheduled to run daily
+   @app.function(
+       secrets=[
+           modal.Secret.from_name("snowflake-secret"),
+           modal.Secret.from_name("postgres-read-replica-prod"),
+       ],
+       # run this pipeline daily at 6:24 AM
+       schedule=modal.Cron("24 6 * * *"),
+       timeout=3000,
+   )
+   def load_table_from_database(
+       table: str,
+       incremental_col: str,
+       dev: bool = False,
+   ) -> None:
+       # Placeholder for future logic
+       pass
+   ```
+
+4. Write your `dlt` pipeline:
```

Comment:
Where the user should put the code from this section? Is it still goes to `sql_database_pipeline.py`?

Reply:
It goes to `sql_database_pipeline.py`